packages/react-storage/src/components/StorageBrowser/Views/LocationsView/Controls/DataTable.tsx

```diff
@@ -0,0 +1,185 @@
+import React from 'react';
+
+import {
+  DataTable,
+  TABLE_DATA_BUTTON_CLASS,
+  TABLE_HEADER_BUTTON_CLASS_NAME,
+  TABLE_HEADER_CLASS_NAME,
+} from '../../../components/DataTable';
+import { useControl } from '../../../context/controls';
+import { useLocationsData } from '../../../context/actions';
+import { LocationAccess, Permission } from '../../../context/types';
+import { compareStrings } from '../../../context/controls/Table';
+import { ButtonElement, IconElement } from '../../../context/elements';
+
+export type SortDirection = 'ascending' | 'descending' | 'none';
+
+export type SortState<T> = {
+  selection: keyof T;
+  direction: SortDirection;
+};
+
+const LocationsViewColumnSortMap = {
+  scope: compareStrings,
+  type: compareStrings,
+  permission: compareStrings,
+};
+
+const getLocationsData = ({
+  data,
+  onLocationClick,
+  onTableHeaderClick,
+  sortState,
+}: {
+  data: LocationAccess<Permission>[];
+  onLocationClick: (location: LocationAccess<Permission>) => void;
+  onTableHeaderClick: (payload: keyof LocationAccess<Permission>) => void;
+  sortState: SortState<LocationAccess<Permission>>;
+}) => {
+  const { selection, direction } = sortState;
+
+  const columns = [
+    {
+      children: (
+        <ButtonElement
+          variant="sort"
+          className={TABLE_HEADER_BUTTON_CLASS_NAME}
+        >
+          Name
+          {selection === 'scope' ? (
+            <IconElement
+              variant={
+                direction === 'none'
+                  ? 'sort-indeterminate'
+                  : `sort-${direction}`
+              }
+            />
+          ) : (
+            <IconElement variant="sort-indeterminate" />
+          )}
+        </ButtonElement>
+      ),
+      key: 'th-scope',
+      className: `${TABLE_HEADER_CLASS_NAME} ${TABLE_HEADER_CLASS_NAME}--scope`,
```

Comment:
Adding this line from the previous `TableControl` implementation here: https://github.com/aws-amplify/amplify-ui/blob/8e34f98ac783c2964bf34e3da36d58581599dfd7/packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx#L90-L94